shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1608,6 +1908,96 @@ - (void)setTextInputEditingState:(NSDictionary*)state {
+  _reusableInputView.scribbleFocusing = true;
+  [_textInputDelegate focusElement:elementIdentifier
+                           atPoint:focusReferencePoint
+                            result:^(id _Nullable result) {
+                              _reusableInputView.scribbleFocusing = false;
+                              _reusableInputView.scribbleFocused = true;
+                              completion(_reusableInputView);
+                            }];
+}
+
+- (BOOL)indirectScribbleInteraction:(UIIndirectScribbleInteraction*)interaction
+         shouldDelayFocusForElement:(UIScribbleElementIdentifier)elementIdentifier {
+  return NO;
+}
+
+- (void)indirectScribbleInteraction:(UIIndirectScribbleInteraction*)interaction
+          willBeginWritingInElement:(UIScribbleElementIdentifier)elementIdentifier {
+}
+
+- (void)indirectScribbleInteraction:(UIIndirectScribbleInteraction*)interaction
+          didFinishWritingInElement:(UIScribbleElementIdentifier)elementIdentifier {
+}
+
+- (CGRect)indirectScribbleInteraction:(UIIndirectScribbleInteraction*)interaction
+                      frameForElement:(UIScribbleElementIdentifier)elementIdentifier {
+  NSValue* elementValue = [_scribbleElements objectForKey:elementIdentifier];
+  if (elementValue == nil) {
+    return CGRectZero;
+  }
+  return [elementValue CGRectValue];
+}
+
+- (void)indirectScribbleInteraction:(UIIndirectScribbleInteraction*)interaction
+              requestElementsInRect:(CGRect)rect
+                         completion:
+                             (void (^)(NSArray<UIScribbleElementIdentifier>* elements))completion {
+  [_textInputDelegate
+      requestElementsInRect:rect
+                     result:^(id _Nullable result) {
+                       NSMutableArray<UIScribbleElementIdentifier>* elements =
+                           [[NSMutableArray alloc] init];
+                       if ([result isKindOfClass:[NSArray class]]) {
+                         for (NSArray* elementArray in result) {
+                           [elements addObject:elementArray[0]];
+                           [_scribbleElements
+                               setObject:[NSValue valueWithCGRect:CGRectMake(
+                                                                      [elementArray[1] floatValue],
+                                                                      [elementArray[2] floatValue],
+                                                                      [elementArray[3] floatValue],
+                                                                      [elementArray[4] floatValue])]
+                                  forKey:elementArray[0]];
+                         }
+                       }
+                       completion(elements);
+                     }];
+}
+
+#pragma mark - Methods related to Scribble support
+
+- (void)setupIndirectScribbleInteraction:(FlutterViewController*)viewController {
```

Comment:
There already is a public setter for this via the property `setViewController`.  If people are expected to use this instead considering putting this logic in that method or making the property readonly.